packages/examples/fortune/launcher/server/src/plugins/s3.ts

```diff
@@ -0,0 +1,82 @@
+import "dotenv/config";
+import fp from "fastify-plugin";
+import { FastifyPluginAsync } from "fastify";
+import * as Minio from 'minio';
+import { escrow as escrowSchema } from '../schemas/escrow.js';
+import { Type } from "@sinclair/typebox";
+import Ajv from "ajv";
+import { EX_ORACLE_ADDRESS, EX_ORACLE_URL, REC_ORACLE_ADDRESS, REC_ORACLE_URL, REP_ORACLE_ADDRESS, REP_ORACLE_URL } from "../constants/oracles.js";
+
+const ConfigSchema = Type.Strict(
+    Type.Object({
+        S3_HOST: Type.String(),
+        S3_PORT: Type.Number(),
+        S3_ACCESS_KEY: Type.String(),
+        S3_SECRET_KEY: Type.String(),
+        S3_BUCKET_NAME: Type.String(),
+        S3_BASE_URL: Type.String()
+    })
+  );
+  
+  const ajv = new Ajv({
+    allErrors: true,
+    removeAdditional: true,
+    useDefaults: true,
+    coerceTypes: true,
+    allowUnionTypes: true,
+  });
+
+
+class S3Client {
+    private s3Client: Minio.Client;
+    private s3Host = process.env.S3_HOST as string;
+    private s3Port = Number(process.env.S3_PORT);
+    private s3AccessKey = process.env.S3_ACCESS_KEY as string;
+    private s3SecretKey = process.env.S3_SECRET_KEY as string;
+    private  s3BucketName = process.env.S3_BUCKET_NAME as string;
+    private  s3BaseUrl = process.env.S3_BASE_URL as string;
+    constructor() {
+        this.s3Client = new Minio.Client({
```

Comment:
I used Minio because is more generic and allows to use buckets hosted outside aws, with aws s3 client you cannot pass the host of the bucket, just a bucket name, so if for testing or local development we use minio it won't work